core/src/main/java/com/datastax/oss/driver/internal/core/channel/ChannelFactory.java

```diff
@@ -236,7 +259,22 @@ protected void initChannel(Channel channel) {
               .sslHandlerFactory()
               .map(f -> f.newSslHandler(channel, address))
               .map(h -> pipeline.addLast("ssl", h));
+
+          NodeMetricUpdater nodeMetricUpdater;
+          if (node instanceof DefaultNode) {
+            nodeMetricUpdater = ((DefaultNode) node).getMetricUpdater();
+          } else {
+            nodeMetricUpdater = NoopNodeMetricUpdater.INSTANCE;
+          }
           pipeline
+              .addLast(
```

Comment:
driver 3.x selectively excludes initializing these if metrics are disabled, could do something similar here but wasn't sure if it was worth it (and also there isn't a general flag I can check).